gitclone/checkout_helper.go

```diff
@@ -0,0 +1,146 @@
+package gitclone
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/bitrise-io/go-utils/command/git"
+	"github.com/bitrise-io/go-utils/log"
+)
+
+type fetchOptions struct {
+	// Fetch tags ("--tags")
+	tags bool
+	// Clone depth ("--depth=")
+	depth int
+}
+
+func (t fetchOptions) IsFullDepth() bool {
+	return t.depth == 0
+}
+
+const branchRefPrefix = "refs/heads/"
+
+type fetchRef struct {
+	remote, ref string
+}
+
+func newOriginFetchRef(ref string) *fetchRef {
```

Comment:
NOTE: i am not sure if it worth returning a pointer here, this functional always can create a fetchRef and does not store huge properties

Reply:
removed